File: Frontend-POS/RBMS-POS-Client/src/app/core/api/models/shop-settings-response-model.ts
/* tslint:disable */
/* eslint-disable */
import { OperatingHourModel } from '../models/operating-hour-model';
export interface ShopSettingsResponseModel {
  address?: string | null;
  companyNameEnglish?: string | null;
  companyNameThai?: string | null;
  createdAt?: string;
  createdByName?: string | null;
  description?: string | null;
  facebook?: string | null;
  foodType?: string | null;
  hasTwoPeriods?: boolean;
  instagram?: string | null;
  lineId?: string | null;
  logoFileId?: number | null;
  logoFileName?: string | null;
  operatingHours?: Array<OperatingHourModel> | null;
  paymentQrCodeFileId?: number | null;
  paymentQrCodeFileName?: string | null;
  phoneNumber?: string | null;
  shopEmail?: string | null;
  shopNameEnglish?: string | null;
  shopNameThai?: string | null;
  shopSettingsId?: number;
  taxId?: string | null;
  updatedAt?: string | null;
  updatedByName?: string | null;
  website?: string | null;
}

File: Backend-POS/POS.Main/POS.Main.Business.Admin/Models/ShopSettings/ShopSettingsResponseModel.cs
using POS.Main.Core.Enums;

namespace POS.Main.Business.Admin.Models.ShopSettings;

public class ShopSettingsResponseModel
{
    public int ShopSettingsId { get; set; }
    public string ShopNameThai { get; set; } = string.Empty;
    public string ShopNameEnglish { get; set; } = string.Empty;
    public string? CompanyNameThai { get; set; }
    public string? CompanyNameEnglish { get; set; }
    public string TaxId { get; set; } = string.Empty;
    public string FoodType { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int? LogoFileId { get; set; }
    public string? LogoFileName { get; set; }
    public bool HasTwoPeriods { get; set; }
    public string Address { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string? ShopEmail { get; set; }
    public string? Facebook { get; set; }
    public string? Instagram { get; set; }
    public string? Website { get; set; }
    public string? LineId { get; set; }
    public int? PaymentQrCodeFileId { get; set; }
    public string? PaymentQrCodeFileName { get; set; }
    public List<OperatingHourModel> OperatingHours { get; set; } = new();

    // Audit info
    public string? CreatedByName { get; set; }
    public DateTime CreatedAt { get; set; }
    public string? UpdatedByName { get; set; }
    public DateTime? UpdatedAt { get; set; }
}
